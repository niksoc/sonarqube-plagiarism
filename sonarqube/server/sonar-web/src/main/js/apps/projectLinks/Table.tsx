/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { translate } from 'sonar-ui-common/helpers/l10n';
import { orderLinks } from '../../helpers/projectLinks';
import LinkRow from './LinkRow';

interface Props {
  links: T.ProjectLink[];
  onDelete: (linkId: string) => Promise<void>;
}

export default class Table extends React.PureComponent<Props> {
  renderHeader() {
    // keep empty cell for actions
    return (
      <thead>
        <tr>
          <th className="nowrap">{translate('project_links.name')}</th>
          <th className="nowrap width-100">{translate('project_links.url')}</th>
          <th className="thin">&nbsp;</th>
        </tr>
      </thead>
    );
  }

  render() {
    if (!this.props.links.length) {
      return <div className="note">{translate('no_results')}</div>;
    }

    const orderedLinks = orderLinks(this.props.links);

    const linkRows = orderedLinks.map(link => (
      <LinkRow key={link.id} link={link} onDelete={this.props.onDelete} />
    ));

    return (
      <div className="boxed-group boxed-group-inner">
        <table className="data zebra" id="project-links">
          {this.renderHeader()}
          <tbody>{linkRows}</tbody>
        </table>
      </div>
    );
  }
}
