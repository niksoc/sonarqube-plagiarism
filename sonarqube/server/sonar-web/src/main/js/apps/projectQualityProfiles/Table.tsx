/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { groupBy, orderBy } from 'lodash';
import * as React from 'react';
import { translate } from 'sonar-ui-common/helpers/l10n';
import { Profile } from '../../api/quality-profiles';
import ProfileRow from './ProfileRow';

interface Props {
  allProfiles: Profile[];
  profiles: Profile[];
  onChangeProfile: (oldProfile: string, newProfile: string) => Promise<void>;
}

export default function Table(props: Props) {
  const profilesByLanguage = groupBy(props.allProfiles, 'language');
  const orderedProfiles = orderBy(props.profiles, 'languageName');

  // set key to language to avoid destroying of component
  const profileRows = orderedProfiles.map(profile => (
    <ProfileRow
      key={profile.language}
      onChangeProfile={props.onChangeProfile}
      possibleProfiles={profilesByLanguage[profile.language]}
      profile={profile}
    />
  ));

  return (
    <div className="boxed-group boxed-group-inner">
      <table className="data zebra">
        <thead>
          <tr>
            <th className="thin nowrap">{translate('language')}</th>
            <th className="thin nowrap">{translate('quality_profile')}</th>
            {/* keep one empty cell for the spinner */}
            <th>&nbsp;</th>
          </tr>
        </thead>
        <tbody>{profileRows}</tbody>
      </table>
    </div>
  );
}
