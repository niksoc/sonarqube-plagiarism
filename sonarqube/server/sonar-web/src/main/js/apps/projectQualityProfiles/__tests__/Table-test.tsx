/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import Table from '../Table';

it('renders', () => {
  const fooJava = randomProfile('foo-java', 'java');
  const fooJs = randomProfile('foo-js', 'js');
  const allProfiles = [
    fooJava,
    randomProfile('bar-java', 'java'),
    randomProfile('baz-java', 'java'),
    fooJs
  ];
  const profiles = [fooJava, fooJs];
  expect(
    shallow(<Table allProfiles={allProfiles} onChangeProfile={jest.fn()} profiles={profiles} />)
  ).toMatchSnapshot();
});

function randomProfile(key: string, language: string) {
  return {
    activeRuleCount: 17,
    activeDeprecatedRuleCount: 0,
    key,
    name: key,
    language,
    languageName: language,
    organization: 'org'
  };
}
