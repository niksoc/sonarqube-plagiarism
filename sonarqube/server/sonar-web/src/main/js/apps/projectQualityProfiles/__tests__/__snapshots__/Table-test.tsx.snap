// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="boxed-group boxed-group-inner"
>
  <table
    className="data zebra"
  >
    <thead>
      <tr>
        <th
          className="thin nowrap"
        >
          language
        </th>
        <th
          className="thin nowrap"
        >
          quality_profile
        </th>
        <th>
          Â 
        </th>
      </tr>
    </thead>
    <tbody>
      <ProfileRow
        key="java"
        onChangeProfile={[MockFunction]}
        possibleProfiles={
          Array [
            Object {
              "activeDeprecatedRuleCount": 0,
              "activeRuleCount": 17,
              "key": "foo-java",
              "language": "java",
              "languageName": "java",
              "name": "foo-java",
              "organization": "org",
            },
            Object {
              "activeDeprecatedRuleCount": 0,
              "activeRuleCount": 17,
              "key": "bar-java",
              "language": "java",
              "languageName": "java",
              "name": "bar-java",
              "organization": "org",
            },
            Object {
              "activeDeprecatedRuleCount": 0,
              "activeRuleCount": 17,
              "key": "baz-java",
              "language": "java",
              "languageName": "java",
              "name": "baz-java",
              "organization": "org",
            },
          ]
        }
        profile={
          Object {
            "activeDeprecatedRuleCount": 0,
            "activeRuleCount": 17,
            "key": "foo-java",
            "language": "java",
            "languageName": "java",
            "name": "foo-java",
            "organization": "org",
          }
        }
      />
      <ProfileRow
        key="js"
        onChangeProfile={[MockFunction]}
        possibleProfiles={
          Array [
            Object {
              "activeDeprecatedRuleCount": 0,
              "activeRuleCount": 17,
              "key": "foo-js",
              "language": "js",
              "languageName": "js",
              "name": "foo-js",
              "organization": "org",
            },
          ]
        }
        profile={
          Object {
            "activeDeprecatedRuleCount": 0,
            "activeRuleCount": 17,
            "key": "foo-js",
            "language": "js",
            "languageName": "js",
            "name": "foo-js",
            "organization": "org",
          }
        }
      />
    </tbody>
  </table>
</div>
`;
